src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -897,6 +875,12 @@ private async Task CommitCoreAsync(IUIThreadOperationContext operationContext, b
             if (!_renameInfo.TryOnBeforeGlobalSymbolRenamed(_workspace, changedDocumentIDs, this.ReplacementText))
                 return (NotificationSeverity.Error, EditorFeaturesResources.Rename_operation_was_cancelled_or_is_not_valid);
 
+            if (finalSolution.WorkspaceVersion != _workspace.CurrentSolution.WorkspaceVersion)
```

Comment:
Rather than just checking this, move this handling into the if (!TryApplyChanges()) path and you could then call the TryApplyChanges a second time.

Reply:
if it fails for a different reason, do we want to call `TryApplyChanges` twice?

Reply:
TryApplyChanges will return false if you hit the version issue, and will throw for anything else.

Reply:
note: We special case tryApplyChanges returning false in ApplyChangesOperation (https://github.com/dotnet/roslyn/blob/main/src/Workspaces/Core/Portable/CodeActions/Operations/ApplyChangesOperation.cs#L50).  Not sure if that would be helpful here in any way.

The general idea though is that we still attempt to make the change, as long as it's just text changes to files taht themselves haven't been changed.